braindecode/datautil/windowers.py

```diff
@@ -0,0 +1,392 @@
+"""Get epochs from mne.Raw
+"""
+
+# Authors: Hubert Banville <hubert.jbanville@gmail.com>
+#          Lukas Gemein <l.gemein@gmail.com>
+#          Simon Brandt <simonbrandt@protonmail.com>
+#          David Sabbagh <dav.sabbagh@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+import mne
+import pandas as pd
+from ..util import round_list_to_int
+
+
+class BaseWindower(object):
+    """Fixed onset windower
+
+    ToDo: samples or seconds
+
+    Parameters
+    ----------
+    window_size_samples : int | None
+        size of one window in samples
+    overlap_size_samples : int
+        size of overlap for window in samples
+    drop_last_samples : bool
+        Whether to drop the last samples when they don't fit into window. If
+        True, there might be samples at the end of your data that is not used.
+        If False, an additional window will be extracted to include the 
+        remaining samples - this means this additional window will overlap with
+        the previous window.
+
+    """
+
+    def __init__(
+        self,
+        window_size_samples,
+        overlap_size_samples,
+        drop_last_samples,
+            trial_start_offset_samples=0,
+        mapping=None,
+    ):
+        self.window_size_samples = window_size_samples
+        self.overlap_size_samples = overlap_size_samples
+        self.drop_last_samples = drop_last_samples
+        self.trial_start_offset_samples = trial_start_offset_samples
+        self.mapping = mapping
+
+    def __call__(self, raw):
+        return NotImplementedError
+
+
+class FixedLengthWindower(BaseWindower):
+    """Fixed onset windower
+
+    ToDo: samples or seconds
+
+    Parameters
+    ----------
+    window_size_samples : int | None
+        size of one window in samples
+    overlap_size_samples : int
+        size of overlap for window in samples
+    drop_last_samples : bool
```

Comment:
we should pay attention with the word samples as it can be understood as samples for the learning task. Maybe just drop_last or drop_last_times?